persistence/genesis.go

```diff
@@ -2,21 +2,33 @@ package persistence
 
 import (
 	"encoding/hex"
-	"fmt"
 	"log"
 	"math/big"
 
 	"github.com/pokt-network/pocket/persistence/schema"
-	sharedTypes "github.com/pokt-network/pocket/shared/types"
+	"github.com/pokt-network/pocket/shared/types"
 	"github.com/pokt-network/pocket/shared/types/genesis"
-	"github.com/pokt-network/pocket/utility/types"
 )
 
-func (pm *persistenceModule) PopulateGenesisState(state *genesis.GenesisState) { // TODO (Andrew) genericize with actors interface once merged with #111
-	poolValues := make(map[string]*big.Int, 0)
+// TODO(Andrew): generalize with the `actors interface`` once merged with #111
+// WARNING: This function crashes the process if there is an error populating the genesis state.
+func (m *persistenceModule) populateGenesisState(state *genesis.GenesisState) {
+	log.Println("Populating genesis state...")
+
+	// HACK: This is needed to avoid block a previous from interfering with the genesis state hydration
```

Comment:
This is very unclear to me. Populate Genesis should only be called once

Reply:
This is non-actionable, but this comment speaks to how confusing the entire lifecycle issue is to me. I am not clear on the bug nor the solution at this time